src/wallet/rpcwallet.cpp

```diff
@@ -2925,18 +2991,22 @@ UniValue z_getnewaddress(const UniValue& params, bool fHelp)
 
     LOCK2(cs_main, pwalletMain->cs_wallet);
 
+    const CChainParams& chainparams = Params();
+
     EnsureWalletIsUnlocked();
+    EnsureWalletIsBackedUp(chainparams);
 
     auto addrType = defaultType;
     if (params.size() > 0) {
         addrType = params[0].get_str();
     }
 
-    KeyIO keyIO(Params());
+    KeyIO keyIO(chainparams);
     if (addrType == ADDR_TYPE_SPROUT) {
         return keyIO.EncodePaymentAddress(pwalletMain->GenerateNewSproutZKey());
```

Comment:
We cannot cut the next release until this method either generates a deterministic Sprout key, or `z_getnewaddress` returns an error for `ADDR_TYPE_SPROUT`. Either make one of these changes in this PR, or open an issue to do so.

Reply:
https://github.com/zcash/zcash/issues/5368